api/envoy/api/v2/core/base.proto

```diff
@@ -112,7 +146,26 @@ message Node {
   // This is motivated by informing a management server during canary which
   // version of Envoy is being tested in a heterogeneous fleet. This will be set
   // by Envoy in management server RPCs.
-  string build_version = 5;
+  // Deprecated in favor of the user_agent_name and user_agent_version values.
+  string build_version = 5 [deprecated = true];
+
+  // Free-form string that identifies the entity requesting config.
+  // E.g. "envoy" or "grpc"
+  string user_agent_name = 6;
+
+  // Free-form string that identifies the version of the entity requesting config.
+  // E.g. "1.12.2" or "abcd1234", or "SpecialEnvoyBuild"
+  string user_agent_version = 7;
+
+  // Extension major version support
```

Comment:
Maybe should be something like "A list of extensions and their versions supported by the node."

Reply:
done